### ISSUE TITLE
What is the reason for NaiveSyncBatchNorm to only accept float32 tensors?

### ISSUE THREAD

**Issue by tyomj** [open-mmlab/mmdetection3d#2444]
The current restriction, as seen in [this link](https://github.com/open-mmlab/mmdetection3d/blob/4ff136163ee4e6436e7f5c44f73c4a4932aa7657/mmdet3d/models/layers/norm.py#L125-L126), prohibits running any network that includes NaiveSyncBatchNorm with AmpOptimWrapper. Do you know why this is the case? 

Out of curiosity, I commented out these lines and trained a network for a few epochs using both single-GPU and multi-GPU setups with Amp on and off. I noticed a 5% batch time improvement with AMP on Nvidia V100, but no difference in loss.

**Comment by sunjiahao1999** [comment]
Pytorch `BatchNorm` is still `float32` in autocast-enabled regions [link](https://pytorch.org/docs/stable/amp.html#cuda-ops-that-can-autocast-to-float32). So, `NativeSyncBatchNorm` follow this operation. The problem you are getting is caused by the update of Openmmlab 2.0. We are now working on fixing this issue. Thanks for your exploration on this issue, we'll go further to check if `NativeBatchNorm` is stable in `float16`.
